resources/prosody-plugins/token/util.lib.lua

```diff
@@ -205,19 +250,46 @@ function Util:verify_room(session, room_address)
         return true;
     end
 
-    local room = string.match(room_address, "^(%w+)@");
+    -- extract room name using all chars, except the not allowed ones
+    local room,_,_ = jid.split(room_address);
     if room == nil then
         log("error",
             "Unable to get name of the MUC room ? to: %s", room_address);
         return true;
     end
 
     local auth_room = session.jitsi_meet_room;
-    if room ~= string.lower(auth_room) then
-        return false;
+    if not self.enableDomainVerification then
+        if room ~= string.lower(auth_room) then
+            return false;
+        end
+
+        return true;
     end
 
-    return true;
+    local room_address_to_verify = jid.bare(room_address);
+    -- parses bare room address, for multidomain expected format is:
+    -- [subdomain]roomName@conference.domain
+    local target_subdomain, target_room
+            = room_address_to_verify:match("^%[([^%]]+)%](.+)$");
+
+    local auth_domain = session.jitsi_meet_domain;
+    if target_subdomain then
+        -- from this point we depend on muc_domain_base,
+        -- deny access if option is missing
+        if not self.muc_domain_base then
+            module:log("warn", "No 'muc_domain_base' option set, denying access!");
+            return false;
+        end
+
+        return room_address_to_verify == jid.join(
+            "["..auth_domain.."]"..string.lower(auth_room), self.muc_domain);
+    else
+        -- we do not have a domain part (multidomain is not enabled)
```

Comment:
On a system with multidomain enabled you can have both:
[subdomain]roomname@conference.domain and roomname@conference.domain and this is an attempt to support both.
The only difference is that in jwt in option 1 aud claim value will be 'subdomain' and in option 2 it will be 'domain'.